src/Discord.OAuth2/DiscordHandler.cs

```diff
@@ -28,9 +28,9 @@ protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIden
             if (!response.IsSuccessStatusCode)
                 throw new HttpRequestException($"Failed to retrieve Discord user information ({response.StatusCode}).");
 
-            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+            var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
```

Comment:
The new `JsonDocument` type implements `IDisposable`, so you should add a `using` here:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/5a7bdf0d5708fbfe1e9f86e3aa8d4881475f1cca/src/AspNet.Security.OAuth.Discord/DiscordAuthenticationHandler.cs#L51

Reply:
> 
> 
> The new `JsonDocument` type implements `IDisposable`, so you should add a `using` here:
> 
> https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/5a7bdf0d5708fbfe1e9f86e3aa8d4881475f1cca/src/AspNet.Security.OAuth.Discord/DiscordAuthenticationHandler.cs#L51

I hope it's OK to use the root element (`JsonElement`) outside the disposed scope?
See https://github.com/RogueException/Discord.OAuth2/pull/12/commits/a3177e16299f0702b0a851e31bae0b340eeff853

Reply:
I don’t think so. It should be disposed when you’ve finished reading anything from it. If you look at the Microsoft OAuth providers for 3.0 they do the same as the code in the link I posted.